typescript/sdk/src/consts/chainMetadata.ts

```diff
@@ -87,36 +87,71 @@ export interface RpcPaginationOptions {
  * Keep in sync with ChainMetadata above
  */
 export const ChainMetadataSchema = z.object({
-  chainId: z.number().positive(),
-  domainId: z.number().positive().optional(),
-  name: z.string(),
-  displayName: z.string().optional(),
-  displayNameShort: z.string().optional(),
+  chainId: z.number().positive().describe(`The chainId of the chain.`),
+  domainId: z
+    .number()
+    .positive()
+    .optional()
+    .describe(
+      'The domainId of the chain, should generally default to `chainId`. Consumer of `ChainMetadata` should use this value if present, but otherwise fallback to `chainId`.',
+    ),
+  name: z
+    .string()
+    .describe(
+      `The string identifier of the chain, used as the key in dictornaries.`,
+    ),
+  displayName: z
+    .string()
+    .optional()
+    .describe(`Human-readable name of the chain.`),
+  displayNameShort: z
+    .string()
+    .optional()
+    .describe(`Shorter human-readable name of the chain.`),
   nativeToken: z
     .object({
       name: z.string(),
       symbol: z.string(),
       decimals: z.number().positive(),
     })
     .optional(),
-  publicRpcUrls: z
+  rpcUrls: z
     .array(
       z.object({
-        http: z.string().url(),
-        webSocket: z.string().optional(),
+        http: z.string().url().describe(`The HTTP RPC endpoint URL.`),
```

Comment:
Hmm the problem is that I do think we need to support http without ssl because often times local nodes don't have it IMO. But we can extend the description to recommend HTTPS as much as possible

Reply:
Okay but note, http won't work in the browser (e.g. in the explorer or warp ui) because non-ssl traffic is forbidden in an https context.

Reply:
Yeah noted, we can have stricter validation on the explorer side. I mostly care about the shapes being harmonized